### PR TITLE
Ft gp model

### PR DESCRIPTION
**Pull request by lilly-muyama**
# Spatial-temporal Modelling Endpoint

**_WHAT DOES THIS PR DO?_**
    This PR creates an endpoint that provides spatial-temporal predictions for a heatmap.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
    https://airqoteam.atlassian.net/browse/PLAT-278

**_WHAT IS THE LINK TO THE PR BRANCH_**
    https://github.com/airqo-platform/AirQo-api/blob/ft-gp-model/src/predict/requirements.txt

**_HOW DO I TEST OUT THIS PR?_**
    git clone this repo, then run the following commands:
    `cd src/predict`
    `env\scripts\activate` for windows or `source env/bin/activate` for MacOS
    `pip install -r requirements.txt`
    `flask run`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
   http://127.0.0.1:5000/api/v1/predict/heatmap

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

**Review by Baalmart — APPROVED** [review]
Endpoint functions as expected. Thanks @lilly-muyama 

We just need to find a way of returning 404 errors in JSON format. Example: when one adds a space or "return characters" after the endpoint partial path, you get a 404 in HTML.

**Review by ja-odur — COMMENTED** [review]
@lilly-muyama  I tested the predict/heatmap endpoint and go back results that included longtitude, latitude, mean and variance information. I am wondering if the "mean" is the same as the predicted value by the model otherwise I would expect a predicted value along the returned data.

File: src/predict/app.py [hunk 1]
```diff
@@ -1,24 +1,56 @@
...
+
 _logger = logging.getLogger(__name__)
 
 app = Flask(__name__)
-
-# Allow cross-brower resource sharing
 CORS(app)
+cache.init_app(app)
 
 app.config["MONGO_URI"] = os.getenv("MONGO_URI")
```
Comment (danielogen) [inline src/predict/app.py:49]: I think for uniformity in the immediate future, this line will look like this: `os.getenv("MONGO_GCE_URI")`

File: src/predict/helpers/utils.py [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+from os import makedirs
+from os.path import join, isdir, isfile, basename
+import numpy as np
+import tensorflow as tf
 import requests
 
 MET_API_URL= os.getenv("MET_API_UR")
 MET_API_CLIENT_ID= os.getenv("MET_API_CLIENT_ID")
 MET_API_CLIENT_SECRET =os.getenv("MET_API_CLIENT_SECRET")
+MONGO_URI = os.getenv("MONGO_URI")
```
Comment (danielogen) [inline src/predict/helpers/utils.py:18]: `os.getenv("MONGO_GCE_URI")` - can be updated later

**Review by danielogen — APPROVED** [review]
LGTM

**Comment by lilly-muyama** [comment]
> 
> 
> @lilly-muyama I tested the predict/heatmap endpoint and go back results that included longtitude, latitude, mean and variance information. I am wondering if the "mean" is the same as the predicted value by the model otherwise I would expect a predicted value along the returned data.

@ja-odur the mean is the predicted value. I have changed the response of the endpoint to reflect that

**Review by ja-odur — APPROVED** [review]
LGTM

**Pull request merged**